### PR TITLE
Introduce issue templates

### PR DESCRIPTION
**Pull request by pavelkumbrasev**
### Description 
Introduce minimalistic templates for issues. Templates are ordered by the importance.

### CODE REVIEW

**Comment by isaevil** [comment]
What about #1087?

**Comment by vbm23** [comment]
> What about #1087?

@pavelkumbrasev Is this MR related to #1087?

**Comment by vbm23** [comment]
Hi Team @aleksei-fedotov @omalyshe @aepanchi 
Can you help review this MR in order for us to meet the time line of June 28th for https://github.com/uxlfoundation/open-source-working-group/issues/98?

File: .github/ISSUE_TEMPLATE/1_question.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+name: Ask a question
+about: Use this template for any sort of a question
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/1_question.md:3]:
```suggestion
about: Use this template for any questions
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+name: Report a bug or a performance issue
+about: Use this template to report unexpected behavior
+title: ''
+labels: ''
+assignees: ''
+---
+
+# Summary
+Provide a short summary of the issue. Sections below provide guidance on what
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:10]:
```suggestion
Provide a short summary of the issue. 
See the sections below
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+factors are considered important to reproduce an issue.
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:11]:
```suggestion
for factors important for the reproduction of an issue.
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+# Version
+Report oneTBB version that is used to reproduce the problem.
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:14]:
```suggestion
Report oneTBB version used to reproduce the problem.
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+# Environment
+Please provide any environmental details that you deem significant for reproducing the issue.
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:17]:
```suggestion
Provide any environmental details that you consider significant for reproducing the issue.
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+The following information could be important:
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:18]:
```suggestion
The following information is important:
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+* HW you use
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:19]:
```suggestion
* Hardware
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+* OS name and version
+* Compiler version
+
+# Observed behavior
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:23]:
```suggestion
# Observed Behavior
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+Document behavior you observe.
+
+# Expected behavior
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:26]:
```suggestion
# Expected Behavior
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+Document behavior you expect.
+
+# Steps to reproduce
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:29]:
```suggestion
# Steps To Reproduce
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+Please check that the issue is reproducible with the latest revision on
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:30]:
```suggestion
Check that the issue is reproducible with the latest revision
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+master. Include all the steps to reproduce the issue.
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:31]:
```suggestion
of the master branch. Include all the steps to reproduce the issue.
```

File: .github/ISSUE_TEMPLATE/3_feature_request.md [hunk 13]
```diff
@@ -0,0 +1,17 @@
+---
+name: Request a feature
+about: Use this template to request new functionality or change the behavior of the library
+title: ''
+labels: ''
+assignees: ''
+---
+
+# Summary
+Include a short summary of the request. Sections below provide guidance on
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/3_feature_request.md:10]:
```suggestion
Include a short summary of the request. 

See the sections below
```

File: .github/ISSUE_TEMPLATE/3_feature_request.md [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+what factors are considered important for a feature request.
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/3_feature_request.md:11]:
```suggestion
for factors important for a feature request.
```

File: .github/ISSUE_TEMPLATE/3_feature_request.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+
+# Problem statement
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/3_feature_request.md:13]:
```suggestion
# Problem Statement
```

File: .github/ISSUE_TEMPLATE/3_feature_request.md [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+Describe the problem you are trying to solve with reasonable level of details.
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/3_feature_request.md:14]:
```suggestion
Describe the problem you want to solve with a reasonable level of detail.
```

File: .github/ISSUE_TEMPLATE/3_feature_request.md [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+
+# Preferred solution
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/3_feature_request.md:16]:
```suggestion
# Preferred Solution
```

File: .github/ISSUE_TEMPLATE/3_feature_request.md [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+Document your thoughts on what solution may look like.
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/3_feature_request.md:17]:
```suggestion
Provide your ideas regarding problem solutions.
```

File: .github/ISSUE_TEMPLATE/4_documentation.md [hunk 19]
```diff
@@ -0,0 +1,19 @@
+---
+name: Request a documentation change
+about: Use this template to report documentation issue or request documentation changes
+title: ''
+labels: 'documentation'
+assignees: ''
+---
+
+# Summary
+Include a short summary of the issue or request. Sections below provide
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/4_documentation.md:10]:
```suggestion
Include a short summary of the issue or request. 
See the sections below
```

File: .github/ISSUE_TEMPLATE/4_documentation.md [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+guidance on what factors are considered important for a documentation
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/4_documentation.md:11]:
```suggestion
for factors important for a documentation
```

File: .github/ISSUE_TEMPLATE/4_documentation.md [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+issue.
+
+# URLs
+Include pointers to documents that are impacted.
+
+# Additional details
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/4_documentation.md:17]:
```suggestion
# Additional Details
```

File: .github/ISSUE_TEMPLATE/4_documentation.md [hunk 22]
```diff
@@ -0,0 +1,19 @@
...
+Provide detailed description of the expected changes in documentation
```
Comment (aepanchi) [inline .github/ISSUE_TEMPLATE/4_documentation.md:18]:
```suggestion
Provide a detailed description of the expected changes in documentation
```

File: .github/ISSUE_TEMPLATE/2_bug_report.md [hunk 23]
```diff
@@ -0,0 +1,31 @@
+---
+name: Report a bug or a performance issue
+about: Use this template to report unexpected behavior
+title: ''
+labels: ''
```
Comment (pavelkumbrasev) [inline .github/ISSUE_TEMPLATE/2_bug_report.md:5]:
```suggestion
labels: 'bug'
```

File: .github/ISSUE_TEMPLATE/3_feature_request.md [hunk 24]
```diff
@@ -0,0 +1,17 @@
+---
+name: Request a feature
+about: Use this template to request new functionality or change the behavior of the library
+title: ''
+labels: ''
```
Comment (pavelkumbrasev) [inline .github/ISSUE_TEMPLATE/3_feature_request.md:5]:
```suggestion
labels: 'new feature'
```

**Comment by pavelkumbrasev** [comment]
> > What about #1087?
> 
> @pavelkumbrasev Is this MR related to #1087?

Not really. I did not remember about #1087 when preparing this PR. We can proceed with either of these PRs. I'd prefer this option as more lite weight.

**Comment by vbm23** [comment]
> > > What about #1087?
> > 
> > 
> > @pavelkumbrasev Is this MR related to #1087?
> 
> Not really. I did not remember about #1087 when preparing this PR. We can proceed with either of these PRs. I'd prefer this option as more lite weight.

Noted @pavelkumbrasev and makes sense for now though as part of continuous improvement you can always improve in near future if needed.

**Review by omalyshe — APPROVED** [review]
LGTM.

**Pull request merged**